uaa/src/test/java/org/cloudfoundry/identity/uaa/mock/zones/OrchestratorZoneControllerTransactionRollbackTests.java

```diff
@@ -0,0 +1,126 @@
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ImportResource;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.context.support.PropertySourcesPlaceholderConfigurer;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.context.web.WebAppConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.ResultMatcher;
+import org.springframework.test.web.servlet.request.MockHttpServletRequestBuilder;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.util.StringUtils;
+import org.springframework.web.context.WebApplicationContext;
+
+@ExtendWith(SpringExtension.class)
+@ExtendWith(PollutionPreventionExtension.class)
+@ActiveProfiles("default")
+@WebAppConfiguration
+@ContextConfiguration(classes = OrchestratorZoneTestConfiguration.class)
+public class OrchestratorZoneControllerTransactionRollbackTests {
+
+    private static final String ZONE_NAME = "The Twiglet Zone";
+    private static final String SUB_DOMAIN_NAME = "sub-domain-01";
+    private static final String ADMIN_CLIENT_SECRET = "admin-secret-01";
+
+    private MockMvc mockMvc;
+    private ApplicationContext applicationContext;
+
+    @BeforeEach
+    void setUp(@Autowired MockMvc mockMvc, @Autowired ApplicationContext applicationContext) {
+        this.mockMvc = mockMvc;
+        this.applicationContext = applicationContext;
+    }
+
+    @Test
+    void testCreateZone_TransactionRollBack() throws Exception {
+
+        BeanDefinitionRegistry registry = null;
+        BeanDefinition originalBeanDefinition = null;
+        try {
+            AutowireCapableBeanFactory factory = applicationContext.getAutowireCapableBeanFactory();
+            registry = (BeanDefinitionRegistry) factory;
+            originalBeanDefinition = registry.getBeanDefinition("identityProviderProvisioning");
+
+            GenericBeanDefinition genericBeanDefinition = new GenericBeanDefinition();
+            genericBeanDefinition.setBeanClass(MockIdentityProviderProvisioning.class);
+            registry.removeBeanDefinition("identityProviderProvisioning");
+
+            registry.registerBeanDefinition("identityProviderProvisioning", genericBeanDefinition);
+
+            OrchestratorZoneRequest orchestratorZoneRequest = getOrchestratorZoneRequest(ZONE_NAME, ADMIN_CLIENT_SECRET,
+                                                                                         SUB_DOMAIN_NAME);
+
+            MvcResult result = mockMvc.perform(post("/orchestrator/zones").contentType(APPLICATION_JSON).content(
+                JsonUtils.writeValueAsString(orchestratorZoneRequest))).andReturn();
```

Comment:
Actually, right now this test relies on two exception, first one exception while adding zone admin client, second one registering mock bean,
genericBeanDefinition.setBeanClass(MockIdentityProviderProvisioning.class);

I just commented locally zone admin validation and it is working second exception(we have to test this after merging).

The test right now is checking both identity and orchestrator zone is rollbacked because while retrieve we are using inner join also I am mocking Identity provider so it also won't be created.

The pending users-groups and zone admin client cannot be tested without zoneId since we cannot retrieve zone id created inside create orchestrator zone.